gift_card/models/account_move.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2021 Akretion (<http://www.akretion.com>).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # copy from sale_usability (avoiding unwanted dependencies)
+    sale_ids = fields.Many2many(
+        "sale.order", string="Sale Orders", compute="_compute_sale_ids", store=True
+    )
+
+    @api.depends("invoice_line_ids.sale_line_ids")
+    def _compute_sale_ids(self):
+        for invoice in self:
+            if invoice.move_type == "out_invoice":
+                sales = invoice.invoice_line_ids.mapped("sale_line_ids").mapped(
+                    "order_id"
+                )
+                invoice.sale_ids = sales.ids
+            else:
+                invoice.sale_ids = []
+
+    def action_post(self):
+        res = super().action_post()
+        for move in self:
+            move._create_gift_card()
+        return res
+
+    def _create_gift_card(self):
+        for sale in self.sale_ids:
+            for line in sale.order_line:
+                tmpl = self.env["gift.card.template"].search(
```

Comment:
I recommend to add a gift_cart_template_ids (one2many) on product template in order to avoid a search in the loop.

Reply:
Good idea